### PR TITLE
[Action] symbl_ai -- Add submit video URL 

### PR DESCRIPTION
**Pull request by mjabali**

### CODE REVIEW

**Comment by dylburger** [comment]
- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

**Comment by dannyroosevelt** [comment]
Thanks for submitting the PR, @mjabali! Someone from the team will take a look asap.

File: components/symbl_ai/actions/submit-video-url/post-video-url.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import axios from "axios";
```
Comment (jcortes) [inline components/symbl_ai/actions/submit-video-url/post-video-url.js:1]: If you modify the file `symbl_ai.app.mjs` as I suggested here, then you can replace this line with
```suggestion
import symblAIApp from "../../symbl_ai.app.mjs";
```

File: components/symbl_ai/actions/submit-video-url/post-video-url.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  description: "Submit a Video by providing the URL for processing",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    symbl_ai: {
+      type: "app",
+      app: "symbl_ai",
+      label: "",
+      description: "",
+    },
```
Comment (jcortes) [inline components/symbl_ai/actions/submit-video-url/post-video-url.js:15]: Then you can repace this with
```suggestion
    symblAIApp,
```

File: components/symbl_ai/actions/submit-video-url/post-video-url.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        },
+        data: data,
+      });
+      $.export("$summary", `Conversation Id: ${response.data.conversationId} and Job Id: ${response.data.jobId}`);
+      $.export("conversationId", `${response.data.conversationId}` );
+      $.export("jobId", `${response.data.jobId}` );
+      return response.data;
+    } catch (error) {
+      console.log("Error: ", error);
+    }
```
Comment (jcortes) [inline components/symbl_ai/actions/submit-video-url/post-video-url.js:50]: Also you can replace this block with
```suggestion
    try {
      const response =
        await this.symblAIApp.postVideoUrl({
          $,
          data: {
            url: this.videoUrl,
            name: this.meetingName ?? "",
          },
        });

      $.export("$summary", `Successfully posted video URL. Conversation Id: ${response.conversationId} and Job Id: ${response.jobId}`);

      return response;

    } catch (error) {
      console.log("Error: ", error);
      $.export("$summary", "Failed to post video URL");
    }
```

**Review by jcortes — CHANGES_REQUESTED** [review]
It's looking good. Just a small refactor and it's done. The main idea is to be as consistent as possible with the [folder structure](https://pipedream.com/docs/components/guidelines/#guidelines-patterns) and most of components that are already in the [repo](https://github.com/PipedreamHQ/pipedream/tree/master/components). I think we should make use of the file `symbl_ai.app.mjs` in that way you can separate the concerns on each component. So you can modify this file like this:
```javascript
import { axios } from "@pipedream/platform";

export default {
  type: "app",
  app: "symbl_ai",
  propDefinitions: {},
  methods: {
    getHeaders() {
      return {
        "Content-Type": "application/json",
        "Authorization": `Bearer ${this.$auth.oauth_access_token}`,
      };
    },
    async makeRequest(customConfig) {
      const {
        $,
        path,
        url,
        ...otherConfig
      } = customConfig;

      const basePath = "https://api.symbl.ai/v1";

      const config = {
        url: url ?? `${basePath}${path}`,
        headers: this.getHeaders(),
        ...otherConfig,
      };

      return axios($, config);
    },
    async postVideoUrl({
      $, data,
    }) {
      return this.makeRequest({
        $,
        method: "post",
        path: "/process/video/url",
        data,
      });
    },
  },
};

```

Also can you add the `package.json` file please?
```javascript
{
  "name": "@pipedream/symbl_ai",
  "version": "0.3.3",
  "description": "Pipedream Symbl_ai Components",
  "main": "symbl_ai.app.mjs",
  "keywords": [
    "pipedream",
    "symbl_ai"
  ],
  "homepage": "https://pipedream.com/apps/symbl_ai",
  "author": "Pipedream <support@pipedream.com> (https://pipedream.com/)",
  "license": "MIT",
  "dependencies": {
    "fraudlabspro-nodejs": "^2.1.0"
  },
  "gitHead": "e12480b94cc03bed4808ebc6b13e7fdb3a1ba535",
  "publishConfig": {
    "access": "public"
  }
}

```

**Review by jcortes — COMMENTED** [review]
OK looks good to me now. So about testing your changes there is no unit testing tool just yet as far as I know so what I'm always do is to use the Pipedrean CLI tools are there, you can check the docs [here](https://pipedream.com/docs/cli/reference/#pd-publish) when it comes to publishing an action component

File: components/symbl_ai/package.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    "main": "symbl_ai.app.mjs",
+    "keywords": [
+      "pipedream",
+      "symbl_ai"
+    ],
+    "homepage": "https://pipedream.com/apps/symbl_ai",
+    "author": "Pipedream <support@pipedream.com> (https://pipedream.com/)",
+    "license": "MIT",
+    "dependencies": {
+      "fraudlabspro-nodejs": "^2.1.0",
```
[the remark this answers is not in the document]
Comment (mjabali) [reply components/symbl_ai/package.json]: Thanks! Changes are done.

**Review by jcortes — APPROVED** [review]

**Comment by dannyroosevelt** [comment]
> Hi @jcortes, I have other updates to be submitted but they will be all dependent on this one. Is there any chance we can have this merged soon? Thanks a bunch.

@mjabali to confirm, do these all functionally work for you? If so, we'll get them published asap!

**Comment by mjabali** [comment]
> 

Thanks @dannyroosevelt. We have a whole lot of other actions to be submitted but would like to use this as our starting point for any of the next ones being added.

**Comment by dannyroosevelt** [comment]
Merged and published!

**Pull request merged**